htdocs/js/AchievementList/achievementlist.js

```diff
@@ -0,0 +1,60 @@
+(() => {
+	// Action form validation.
+	const is_achievement_selected = () => {
+		for (const achievement of document.getElementsByName('selected_achievements')) {
+			if (achievement.checked) return true;
+		}
+		document.getElementById('select_achievement_err_msg')?.classList.remove('d-none');
+		document.getElementById('achievement-table')?.addEventListener('change', e => {
+			document.getElementById('select_achievement_err_msg')?.classList.add('d-none');
+		}, { once : true });
```

Comment:
The variable `e` is not used here.  So this should be
```suggestion
		document.getElementById('achievement-table')?.addEventListener('change', () => {
			document.getElementById('select_achievement_err_msg')?.classList.add('d-none');
		}, { once : true });
```
This is done several times in the javascript changes that you have made.  So make this change in those places as well.